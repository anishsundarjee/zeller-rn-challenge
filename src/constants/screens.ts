export const SCREENS = {
  USER_LIST: 'USER_LIST',
  USER_DETAILS: 'USER_DETAILS',
}