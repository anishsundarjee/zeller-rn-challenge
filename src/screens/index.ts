export * from './userList';
export * from './userDetail';