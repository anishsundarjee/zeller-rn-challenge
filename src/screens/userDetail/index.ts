import UserDetailPresenter from './userDetailPresenter';

export { UserDetailPresenter };