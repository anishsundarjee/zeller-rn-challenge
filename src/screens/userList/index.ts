import UserListPresenter from './userListPresenter';

export { UserListPresenter };