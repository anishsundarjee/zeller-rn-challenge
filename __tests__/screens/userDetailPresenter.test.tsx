import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { it, describe, beforeEach } from '@jest/globals';
import { UserDetailPresenter } from '@/screens';
jest.useFakeTimers();

jest.mock('react-native-safe-area-context', () => {
  return {
    useSafeAreaInsets: () => ({ top: 0, bottom: 0 }),
  };
});

const mockedNavigation = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => ({
      goBack: mockedNavigation,
    }),
  };
});

describe('UserDetailPresenter', () => {
  beforeEach(() => {
    mockedNavigation.mockClear();
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <UserDetailPresenter />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('calls goBack when back button is pressed', () => {
    const tree = renderer.create(
      <UserDetailPresenter />
    ).root;
    const backButton = tree.findByProps({ testID: 'back-btn' });
    backButton.props.onPress();
    expect(mockedNavigation).toHaveBeenCalled();
  });
});
