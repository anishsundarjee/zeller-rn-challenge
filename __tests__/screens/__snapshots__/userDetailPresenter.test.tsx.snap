// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailPresenter renders correctly 1`] = `
<View
  style={
    [
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      [
        {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "justifyContent": "space-between",
        },
        {
          "paddingTop": 0,
        },
        {},
      ],
    ]
  }
  testID="user-detail-container"
>
  <View
    style={
      [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        [
          {
            "paddingBottom": 24,
            "paddingTop": 16,
          },
          {
            "paddingHorizontal": 24,
          },
          {},
        ],
      ]
    }
    testID="container-main-view"
  >
    <View
      style={
        [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "column",
            "height": "100%",
          },
        ]
      }
    >
      <View
        style={
          [
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "display": "flex",
              "flexGrow": 1,
              "width": "100%",
            },
          ]
        }
      >
        <View
          row={true}
          style={
            [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {},
              {},
              {
                "flexDirection": "row",
              },
              {
                "alignItems": "flex-start",
                "alignSelf": "center",
                "height": 40,
                "width": 702,
              },
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID="back-btn"
          >
            <Text />
            <View
              style={
                [
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  [
                    {
                      "backgroundColor": "transparent",
                      "height": 8,
                      "width": 8,
                    },
                    undefined,
                  ],
                ]
              }
              testID="padder-view"
            />
            <Text
              body={true}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "color": "#20303C",
                    "writingDirection": "ltr",
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  {},
                  {
                    "textAlign": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              Back
            </Text>
          </View>
          <View
            center={true}
            style={
              [
                undefined,
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              h5={true}
              marginR-s8={true}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "color": "#20303C",
                    "writingDirection": "ltr",
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  {
                    "marginRight": 32,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              User Detail
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
